Tests/TuistKitTests/Generator/WorkspaceGeneratorTests.swift

```diff
@@ -91,19 +111,45 @@ final class WorkspaceGeneratorTests: XCTestCase {
     }
 
     func test_generate_relativeToGroup() throws {
-        try fileHandler.touch(path.appending(component: "Workspace.swift"))
-
         // Given the workspace manifest file exists
         // When generating the workspace
         // I expect the file to be relative to the workspace
 
-        try subject.generate(path: path, graph: graph, options: GenerationOptions())
+        try fileHandler.touch(path.appending(component: "Workspace.swift"))
+        try fileHandler.touch(path.appending(component: "Project.swift"))
+
+        let testWorkspace = Workspace.test(name: "test", contents: [
+            .file(path: path.appending(component: "Workspace.swift")),
+            .file(path: path.appending(component: "Setup.swift")),
+            .project(path: path),
+        ])
+
+        try subject.generate(workspace: testWorkspace, path: path, graph: graph, options: GenerationOptions())
 
         let workspace = try readWorkspace(at: path)
 
         XCTAssertTrue(workspace.contains("location = \"group:Workspace.swift\">"))
     }
 
+    func test_generate_withNestedGroups() throws {
+        // Given the workspace specifies groups
+        // When generating the workspace
+        // I expect the nested groups to be in the correct order (Outer -> Inner -> Project)
+
+        let testWorkspace = Workspace.test(name: "test", contents: [
+            .group(name: "Outer", contents: [.group(name: "Inner", contents: [.project(path: path)])]),
+        ])
+
+        try fileHandler.touch(path.appending(component: "Workspace.swift"))
+        try fileHandler.touch(path.appending(component: "Project.swift"))
+
+        try subject.generate(workspace: testWorkspace, path: path, graph: graph, options: GenerationOptions())
+
+        let workspace = try readWorkspace(at: path)
+
+        XCTAssertTrue(workspace.contains("<Group\n      location = \"container:\"\n      name = \"Outer\">\n      <Group\n         location = \"container:\"\n         name = \"Inner\">\n         <FileRef\n            location = \"group:Project.xcodeproj\">"))
```

Comment:
Same comment here. I'd rather not assert equality comparing the whole string. We have done that in other areas of the project, like serialization, but I think it's not a good practice. In this case you can read the workspace using `xcodeproj` and run the assertions on the workspace children.

Reply:
👍 I agree, it was a quick dirty one to ensure I don't regress when making changes